dask/utils.py

```diff
@@ -381,31 +381,38 @@ def wrapper(func):
 
         return wrapper(func) if func is not None else wrapper
 
-    def __call__(self, arg):
-        # Fast path with direct lookup on type
+    def dispatch(self, cls):
+        """Return the function implementation for the given ``cls``"""
+        # Fast path with direct lookup on cls
         lk = self._lookup
-        typ = type(arg)
         try:
-            impl = lk[typ]
+            impl = lk[cls]
         except KeyError:
             pass
         else:
-            return impl(arg)
+            return impl
         # Is a lazy registration function present?
-        toplevel, _, _ = typ.__module__.partition('.')
+        toplevel, _, _ = cls.__module__.partition('.')
         try:
             register = self._lazy.pop(toplevel)
         except KeyError:
             pass
         else:
             register()
-            return self(arg)  # recurse
+            return self.dispatch(cls) # recurse
         # Walk the MRO and cache the lookup result
-        for cls in inspect.getmro(typ)[1:]:
+        for cls in inspect.getmro(cls)[1:]:
```

Comment:
I should have also seen that. I am guessing since `inspect.getmro` returned its result it somehow didn't conflict with the loop. (so no test needed?)

thanks!